blockchain/state_manager/src/lib.rs

```diff
@@ -660,69 +660,71 @@ where
     where
         CB: FnMut(&Cid, &ChainMessage, &ApplyRet) -> Result<(), anyhow::Error> + Send,
     {
-        span!("compute_tipset_state", {
-            let block_headers = tipset.blocks();
-            let first_block = block_headers
-                .first()
-                .ok_or_else(|| Error::Other("Empty tipset in compute_tipset_state".to_string()))?;
-
-            let check_for_duplicates = |s: &BlockHeader| {
-                block_headers
-                    .iter()
-                    .filter(|val| val.miner_address() == s.miner_address())
-                    .take(2)
-                    .count()
-            };
-            if let Some(a) = block_headers.iter().find(|s| check_for_duplicates(s) > 1) {
-                // Duplicate Miner found
-                return Err(Error::Other(format!("duplicate miner in a tipset ({})", a)));
-            }
-
-            let parent_epoch = if first_block.epoch() > 0 {
-                let parent_cid = first_block
-                    .parents()
-                    .cids()
-                    .get(0)
-                    .ok_or_else(|| Error::Other("block must have parents".to_string()))?;
-                let parent: BlockHeader = self
-                    .blockstore()
-                    .get_obj(parent_cid)?
-                    .ok_or_else(|| format!("Could not find parent block with cid {parent_cid}"))?;
-                parent.epoch()
-            } else {
-                Default::default()
-            };
+        tracing::info_span!("compute_tipset_state")
```

Comment:
this is kind of a downgrade of readability, exposing lambdas gives the reader a needless mental overhead. Can we contrive a macro to hide this implementation detail?

Reply:
Thanks, so much better now!